x-pack/plugins/fleet/server/services/epm/elasticsearch/transform/install.ts

```
@@ -117,6 +134,218 @@ export const installTransform = async (
+        assetsToRemove: previousInstalledTransformEsAssets,
+      }
+    );
+
+    await Promise.all(
+      destinationIndexTemplates
+        .map((destinationIndexTemplate) => {
+          const mergedTemplate = mergeIndexTemplateWithMappings(
+            destinationIndexTemplate.template,
+            transformsSpecifications
+              .get(destinationIndexTemplate.transformModuleId)
+              ?.get('mappings')
+          );
+          if (mergedTemplate !== undefined) {
+            return installComponentAndIndexTemplateForDataStream({
+              esClient,
+              logger,
+              componentTemplates: {},
+              indexTemplate: {
+                templateName: destinationIndexTemplate.installationName,
+                // @ts-expect-error Index template here should not contain data_stream property
+                // as this template is applied to only an index and not a data stream
+                indexTemplate: {
+                  template: mergedTemplate,
+                  priority: 250,
+                  index_patterns: [
+                    transformsSpecifications
+                      .get(destinationIndexTemplate.transformModuleId)
+                      ?.get('destinationIndex').index,
+                  ],
+                  _meta: destinationIndexTemplate._meta,
+                  composed_of: [],
+                },
+              },
+            });
+          }
+        })
+        .filter((p) => p !== undefined)
+    );
+    await Promise.all(
+      transforms.map(async (transform) => {
+        const index = transform.content.dest.index;
+        const pipelineId = transform.content.dest.pipeline;
+
+        const indexExist = await esClient.indices.exists({
+          index,
+        });
+        if (indexExist !== true) {
+          return esClient.indices.create({
+            index,
+            ...(pipelineId ? { settings: { default_pipeline: pipelineId } } : {}),
+          });
+        }
```

Could we avoid a round trip here by always attempting to create the index and ignoring errors if it already exists?

---

Updated here [`8fa543f` (#140046)](https://github.com/elastic/kibana/pull/140046/commits/8fa543f9bd9cc6f331a3f0458fc7354b311ff3cd)